util/systemd/systemd_linux.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build linux
+
+package systemd
+
+import (
+	"log"
+	"sync"
+
+	"github.com/mdlayher/sdnotify"
+)
+
+// XXX(Xe): this may not be the best idea, but this should only be used spairingly as
+// is and Go doesn't have the concept of OS-specific struct members.
+var (
+	systemdLock  sync.Mutex
```

Comment:
sdnotify uses a UNIX socket net.Conn, and net.Conn implementations must be concurrency safe, so this lock shouldn't be necessary.